src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthorizationHeaderFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+package org.opensearch.security.http;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.message.BasicHeader;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.authtoken.jwt.JwtVendor;
+
+import static java.util.Objects.requireNonNull;
+
+class OnBehalfOfJwtAuthorizationHeaderFactory {
```

Comment:
Remove this class, this is acting like a mock for unit tests, any tests that depend on this class should be converted to unit tests where they are faster to run / debug

Reply:
Should we keep this for some of the unhappy cases?

Reply:
You can either mock just mock the unhappy cases you want. We don't actually care about the creation of the tokens for the unit tests, just the handling so you can skip that step and avoid the computation involved in using an actual token.

Reply:
Ya I have removed it since most of them were in the unit testing of this authc backend.